impl/relredis.go

```
@@ -184,22 +187,18 @@ func (r *ReliableRedisStreamClient) Done() {
 	ctx := context.Background()
 
 	for streamName := range r.streamLocks {
-		if err := r.DoneDataStream(ctx, streamName); err != nil {
-			log.Println("error", err, " occured while marking stream", streamName, " as done; moving on to other streams ...")
-		}
+		r.DoneDataStream(ctx, streamName)
 	}
 
 	// release resources
 	r.cleanup()
 }
 
 func (r *ReliableRedisStreamClient) cleanup() {
-	close(r.lbsChan)
+	r.safeCloseLBS()
+
+	r.pubSub.Close()
```

Error should be propogated imo instead of silently ignoring.